legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-duckdb/src/main/java/org/finos/legend/engine/persistence/components/relational/duckdb/jdbc/DuckDBJdbcHelper.java

```diff
@@ -0,0 +1,132 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.persistence.components.relational.duckdb.jdbc;
+
+import org.finos.legend.engine.persistence.components.executor.RelationalExecutionHelper;
+import org.finos.legend.engine.persistence.components.executor.TypeMapping;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.Dataset;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.DatasetDefinition;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.Field;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.FieldType;
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.SchemaDefinition;
+import org.finos.legend.engine.persistence.components.relational.jdbc.JdbcHelper;
+import org.finos.legend.engine.persistence.components.relational.sql.JdbcPropertiesToLogicalDataTypeMapping;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.JDBCType;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class DuckDBJdbcHelper extends JdbcHelper
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(DuckDBJdbcHelper.class);
+
+    public static DuckDBJdbcHelper of(Connection connection)
+    {
+        return new DuckDBJdbcHelper(connection);
+    }
+
+    private DuckDBJdbcHelper(Connection connection)
+    {
+        super(connection);
+    }
+
+    @Override
+    public boolean doesTableExist(Dataset dataset)
+    {
+        try
+        {
+            String name = dataset.datasetReference().name().orElseThrow(IllegalStateException::new);
+            String database = dataset.datasetReference().database().orElse(null);
+            String schema = dataset.datasetReference().group().orElse(null);
+            ResultSet result = this.connection.getMetaData().getTables(database, schema, name, null);
+            return result.next(); // This method returns true if ResultSet is not empty
+        }
+        catch (SQLException e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Dataset constructDatasetFromDatabase(Dataset dataset, TypeMapping typeMapping, boolean escape)
+    {
+        String tableName = dataset.datasetReference().name().orElseThrow(IllegalStateException::new);
+        String schemaName = dataset.datasetReference().group().orElse(null);
+        String databaseName = dataset.datasetReference().database().orElse(null);
+
+        String escapedTableName = tableName.replace("_", "\\_");
```

Comment:
Is this needed in Duck DB ?

Reply:
Have removed this part from this class for now - will properly clean up the whole logic for escaping for snowflake in a separate PR